code/game/objects/items/devices/flash.dm

```diff
@@ -40,6 +40,12 @@
 				overcharged = TRUE
 				add_overlay("overcharge")
 
+//Check if the user is blind
+/obj/item/flash/proc/blind_check(mob/user)
+	if(user && !user.hasTrait("blind")) {
+		to_chat(user, "<span class='notice'>You do not feel anything from the [src] as you are blind.</span>")
+	}
+
```

Comment:
A few things, you're adding a new proc but not calling it, which does nothing.

it's `HAS_TRAIT(user, TRAIT_BLIND)` and please do not use curly brackets with if statements, it's against our contributor guidelines/coding standards.

I don't believe that you tested this because this wouldn't even compile.